### DISCUSSION TITLE
Enhancement Inquiry: Modifying showMatchesPosition Behavior in Meilisearch

### DISCUSSION THREAD

**Discussion by dpcks666** [meilisearch/product#706] [Feedback & Feature Proposal]
Hi

I trust this message finds you well. I'm interested in discussing potential enhancements to the showMatchesPosition functionality in Meilisearch.

Calculation of Starting Point Using attributesToCrop:

Currently, showMatchesPosition is calculated from the original starting point. I'm exploring the possibility of enhancing this feature to allow the calculation of the starting point based on the content cropped by attributesToCrop. Is it feasible to implement such functionality to determine the starting point from the cropped content?

Unit of Calculation for showMatchesPosition:

The existing showMatchesPosition calculates positions in bytes. Would it be possible to extend this functionality to calculate positions on a per-character basis instead of bytes?

Additionally, I'd like to note that implementing HTML-based highlighting for mobile devices can be a challenging task. Are there any specific considerations or recommendations you have for achieving this on mobile platforms?

I am eager to contribute to the implementation of these features and would appreciate your insights on the technical feasibility and any recommended steps to proceed. Your guidance on this matter would be highly valuable.

Thank you for your time and consideration.

**Comment by ManyTheFish** [comment]
Hello @dpcks666,
thank you for your detailed message.
I agree with you that highlighting/cropping is smarter than the showMatchesPosition, and it would be better if we could have "smart matches positions" showing only the best matches and not the complete list of matches.
However, we couldn't change the starting point of the match positions so easily; technically, it's simple to change, but this change breaks Meilisearch because we return less information than before.
That been said, there are several possible alternate solutions we could implement:
1) we could add a new search parameter or index setting to change the `showMatchesPosition` strategy, with 3 possible variants:
    a) all (default): current behavior, Meilisearch returns all the matching positions
    b) bests: show only the bests MatchesPosition based on the internal highlighting score
    c) cropped: show only the MatchesPosition contained in the crop window.
2) we could publish the internal highlighting score for each match, this way, the Meilisearch user could choose which matches they want to keep/show or not.

About the Unit of Calculation for showMatchesPosition, it's the same, it would be breaking to replace the unit, but we could easily have additional character-based positions in the return value.

> 🗒️ Poke @meilisearch/product-team on the upper improvement suggestions.

> Additionally, I'd like to note that implementing HTML-based highlighting for mobile devices can be a challenging task. Are there any specific considerations or recommendations you have for achieving this on mobile platform

I'm not the best to answer this question 😞, but I suggest joining our [discord to ask for help](https://discord.gg/meilisearch), I'm sure you'll find an answer from a mobile developer in our community. 

I hope my answer is clear enough, but don't hesitate to ask more questions or precisions if you need them!

Thanks again for your detailed message,
